README.md

```diff
@@ -31,81 +31,255 @@ yarn add react-native-fs2
 
-#### unlink
+### `unlink`
+```ts
+// unlink(filepath: string): Promise<void>
+await RNFS.unlink('FileToUnlink')
+```
+* Unlinks the item at `filepath`. If the item does not exist, an error will be thrown.
+Also recursively deletes directories (works like Linux `rm -rf`).
 
-#### exists
+### `exists`
+```ts
+// exists(filepath: string): Promise<boolean>
+await RNFS.exists('File')
+```
+* Check if the item exists at `filepath`. If the item does not exist, return false.
 
-#### stopDownload
 
-#### resumeDownload
+### `stopDownload`
+```ts
+// stopDownload(jobId: number): void
+await RNFS.stopDownload('JobID'): void
+```
+* Abort the current download job with this ID. The partial file will remain on the filesystem.
 
-#### isResumable
+### `resumeDownload` (iOS Only)
+```ts
+// resumeDownload(jobId: number): void
+await RNFS.resumeDownload('JobID'): void
+```
+* Resume the current download job with this ID
 
-#### completeHandlerIOS
+### `isResumable` (iOS Only)
+```ts
+// isResumable(jobId: number): Promise<bool>
+if (await RNFS.isResumable('JobID')) {
+    RNFS.resumeDownload('JobID')
+}
+```
+* Check if the the download job with this ID is resumable.
 
-#### readDir
+### `completeHandlerIOS` (iOS Only)
+```ts
+// completeHandlerIOS(jobId: number): void
+await RNFS.completeHandler('JobID')
+```
+*  Tell iOS you are done handling a completed download when using background downloads.
 
-#### readFile
 
-#### read
+### `readDir`
+```ts
+// readDir(dirPath: string): Promise<ReadDirItem[]>
+const dirItems = await RNFS.readDir('DirPath')
+```
+* Retuns an `array` of `ReadDirItem` which are items that are present in the directory
+* `ReadDirItem`
+  * ctime: `Date | undefined` -> The creation date of the file (iOS only)
+  * mtime: `Date | undefined` -> The last modified date of the file
+  * name: `string` -> The name of the item
+  * path: `string` -> The absolute path to the item
+  * size: `number` -> Size in bytes
+  * isFile: () => `boolean` -> Is the file just a file?
+  * isDirectory: () => `boolean` -> Is the file a directory?
+
+
+### `readFile`
+```ts
+// readFile(filepath: string, encodingOrOptions?: EncodingOrOptions): Promise<string>
+const fileData = await RNFS.readFile('DirPath', 'utf8')
+```
+* Reads the `filepath` and return the file `contents`
+* Optionally includes `EncodingOrOptions` with values:
+  * `'utf8'` (default) | `'base64'` (for binary files) | `'ascii'`
+  * ...fileoptions
 
-#### hash
 
-#### writeFile
+### `read`
+```ts
+/*
+  read(
+    filepath: string,
+    length: number = 0,
+    position: number = 0,
+    encoding?: Encoding
+  ): Promise<string> 
+*/
+
+const fileData = await RNFS.read('FileToRead', 0, 0, 'utf8')
+```
+* Reads bytes length from the given position of the file at `filepath` and returns the file `contents`. 
+* Optionally includes `EncodingOrOptions` with values:
```

Comment:
This doc note mentions `EncodingOrOptions`, but example above is just `Encoding` type.